test/base/src/androidTest/kotlin/io/realm/test/shared/EmbeddedObjectTests.kt

```diff
@@ -0,0 +1,1161 @@
+                }
+            )
+        }
+        realm.query<EmbeddedParentWithPrimaryKey>().find().single()
+        assertEquals(2, realm.query<EmbeddedChild>().find().size)
+
+        realm.writeBlocking {
+            copyToRealm(
+                // Need to replicate object as sharing it on native freezes the old one
+                EmbeddedParentWithPrimaryKey().apply {
+                    id = "parent"
+                    child = EmbeddedChild("child3")
+                    childList = realmListOf(EmbeddedChild("child4"), EmbeddedChild("child5"))
+                },
+                MutableRealm.UpdatePolicy.ALL
+            )
+        }
+        realm.query<EmbeddedParentWithPrimaryKey>().find().single()
+        realm.query<EmbeddedChild>().find().run {
+            assertEquals(3, size)
+            forEach { it.id in setOf("child3", "child4", "child5") }
+        }
+    }
+
+    @Test
+    fun copyToRealm_withInitializer() {
+        realm.writeBlocking {
+            copyToRealm(EmbeddedChildWithInitializer())
+        }
+        realm.query<EmbeddedChild>().find().single().run {
+            assertEquals("Initial child", id)
+        }
+    }
+
+    @Test
+    fun setWillDeleteEmbeddedObject() {
+        val parent = realm.writeBlocking {
+            copyToRealm(EmbeddedParent().apply { child = EmbeddedChild() })
+        }
+        realm.query<EmbeddedChild>().find().single()
+        realm.writeBlocking {
+            findLatest(parent)!!.apply {
+                child = null
+            }
+        }
+        realm.query<EmbeddedChild>().find().none()
+    }
+
+    // @org.junit.Test
+    // fun dynamicRealm_settingParentFieldDeletesChild() =
+    //     DynamicRealm.getInstance(realm.configuration).use { realm ->
+    //         realm.executeTransaction {
+    //             val parent = realm.createObject("EmbeddedSimpleParent", "parent")
+    //             val child = realm.createEmbeddedObject("EmbeddedSimpleChild", parent, "child")
+    //
+    //             Assert.assertEquals(1, realm.where("EmbeddedSimpleChild").count())
+    //             parent.setObject("child", null)
+    //             Assert.assertFalse(child.isValid)
+    //             Assert.assertEquals(0, realm.where("EmbeddedSimpleChild").count())
+    //         }
+    //     }
+    //
+
+    @Test
+    fun set_unmanaged() {
+        val parent = realm.writeBlocking {
+            copyToRealm(EmbeddedParent())
+        }
+        realm.query<EmbeddedChild>().find().none()
+        realm.writeBlocking {
+            findLatest(parent)!!.apply {
+                child = EmbeddedChild().apply {
+                    id = "child1"
+                    innerChild = EmbeddedInnerChild()
+                }
+            }
+        }
+        realm.query<EmbeddedChild>().find().single().run {
+            assertEquals("child1", id)
+        }
+        realm.query<EmbeddedChild>().find().single()
```

Comment:
do you mean `EmbeddedInnerChild` here? since you already asserted one instance for `EmbeddedChild`